### PR TITLE
core(deprecations): use inspector issues

### PR DESCRIPTION
**Pull request by connorjclark**
This will fix dbw smoke test.

courtesy of https://bugs.chromium.org/p/chromium/issues/detail?id=1248484

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -218,6 +218,9 @@
     "yargs": "^16.1.1",
     "yargs-parser": "^20.2.4"
   },
+  "resolutions": {
+    "puppeteer/**/devtools-protocol": "0.0.939404"
+  },
```
Comment (connorjclark) [inline package.json:223]: ref: https://github.com/GoogleChrome/lighthouse/pull/12534#issuecomment-846241628

rather add this than upgrade puppeteer in this PR

File: lighthouse-core/audits/deprecations.js [hunk 2]
```diff
@@ -56,16 +56,34 @@ class Deprecations extends Audit {
   static audit(artifacts) {
     const entries = artifacts.ConsoleMessages;
 
-    const deprecations = entries.filter(log => log.source === 'deprecation')
-    // TODO(M91): Temporary ignore until Chrome M91 became stable version.
-    // M91 doesn't throw deprecation on ::-webkit-details-marker.
-    .filter(log => !log.text.includes('::-webkit-details-marker'))
```
Comment (connorjclark) [inline lighthouse-core/audits/deprecations.js:62]: purposefully removed this filter

File: lighthouse-core/audits/deprecations.js [hunk 3]
```diff
@@ -56,16 +56,34 @@ class Deprecations extends Audit {
...
+        .map(deprecation => {
+          return {
+            value: deprecation.message || '',
+            /** @type {LH.Audit.Details.SourceLocationValue} */
+            source: {
+              type: 'source-location',
+              url: deprecation.sourceCodeLocation.url,
+              urlProvider: 'network',
+              line: deprecation.sourceCodeLocation.lineNumber,
+              column: deprecation.sourceCodeLocation.columnNumber - 1,
```
Comment (adamraine) [inline lighthouse-core/audits/deprecations.js:72]: So this is indexed by 1 in CDP? Can you add a comment about that here?

**Review by adamraine — APPROVED** [review]

**Comment by brendankenny** [comment]
can https://github.com/GoogleChrome/lighthouse/blob/68ba77acbc9c22512b93d213b23b05cca443dda4/third-party/chromium-synchronization/inspector-issueAdded-types-test.js#L34-L46 be turned back on now?

File: lighthouse-core/audits/deprecations.js [hunk 4]
```diff
@@ -56,16 +56,34 @@ class Deprecations extends Audit {
...
-    .filter(log => !log.text.includes('::-webkit-details-marker'))
-    .map(log => {
-      return {
-        value: log.text,
-        source: Audit.makeSourceLocationFromConsoleMessage(log),
-      };
-    });
+    let deprecations;
+
+    if (artifacts.InspectorIssues.deprecations?.length) {
```
Comment (brendankenny) [inline lighthouse-core/audits/deprecations.js:61]:
```suggestion
    if (artifacts.InspectorIssues.deprecations.length) {
```
maybe something changed while writing? `deprecations` isn't optional now

File: lighthouse-core/audits/deprecations.js [hunk 5]
```diff
@@ -56,16 +56,34 @@ class Deprecations extends Audit {
...
+            source: {
+              type: 'source-location',
+              url: deprecation.sourceCodeLocation.url,
+              urlProvider: 'network',
+              line: deprecation.sourceCodeLocation.lineNumber,
+              column: deprecation.sourceCodeLocation.columnNumber - 1,
+            },
+          };
+        });
+    } else {
```
Comment (brendankenny) [inline lighthouse-core/audits/deprecations.js:76]: does this need to be an if/else or can these be combined in one go?

File: lighthouse-cli/test/smokehouse/test-definitions/dobetterweb/dbw-expectations.js [hunk 6]
```diff
@@ -303,9 +303,23 @@ const expectations = {
           items: [
             {
               value: /'window.webkitStorageInfo' is deprecated/,
+              source: {
```
Comment (brendankenny) [inline lighthouse-cli/test/smokehouse/test-definitions/dobetterweb/dbw-expectations.js:306]: is sample_v2 unchanged from the old deprecations? That ends up pretty nice

File: lighthouse-core/audits/deprecations.js [hunk 7]
```diff
@@ -56,16 +56,34 @@ class Deprecations extends Audit {
...
+      // Backcompat for <M97.
+      // https://bugs.chromium.org/p/chromium/issues/detail?id=1248484
+      deprecations = entries.filter(log => log.source === 'deprecation')
+        .map(log => {
+          return {
+            value: log.text,
+            source: Audit.makeSourceLocationFromConsoleMessage(log),
+          };
+        });
+    }
```
Comment (brendankenny) [inline lighthouse-core/audits/deprecations.js:86]: needs tests?

File: lighthouse-core/audits/deprecations.js (hunk 5 above)
Comment (connorjclark) [reply lighthouse-core/audits/deprecations.js:76]: There are some canary releases where both would be populated.

**Comment by paulirish** [comment]
> can
> 
> https://github.com/GoogleChrome/lighthouse/blob/68ba77acbc9c22512b93d213b23b05cca443dda4/third-party/chromium-synchronization/inspector-issueAdded-types-test.js#L34-L46
> 
> be turned back on now?

nope. there's like 8 new types we don't handle. but Adam Read is working on this.

**Comment by connorjclark** [comment]
The first skipped test was enabled, b/c it is only meant to let us know when a new type exists.

**Pull request merged**